src/common/timing/nav-timing.js

```diff
@@ -27,44 +27,43 @@ var DOM_CONTENT_LOAD_EVENT = 'domContentLoadedEvent'
 
 export var navTimingValues = []
 
-export function addPT (offset, pt, v) {
+export function addPT (offset, pt, v = {}) {
+  if (!pt) return
   v.of = offset
-  addRel(offset, offset, v, 'n')
-  addRel(pt[UNLOAD_EVENT + START], offset, v, 'u')
-  addRel(pt[REDIRECT + START], offset, v, 'r')
-  addRel(pt[UNLOAD_EVENT + END], offset, v, 'ue')
-  addRel(pt[REDIRECT + END], offset, v, 're')
-  addRel(pt['fetch' + START], offset, v, 'f')
-  addRel(pt[DOMAIN_LOOKUP + START], offset, v, 'dn')
-  addRel(pt[DOMAIN_LOOKUP + END], offset, v, 'dne')
-  addRel(pt['c' + ONNECT + START], offset, v, 'c')
-  addRel(pt['secureC' + ONNECT + 'ion' + START], offset, v, 's')
-  addRel(pt['c' + ONNECT + END], offset, v, 'ce')
-  addRel(pt[REQUEST + START], offset, v, 'rq')
-  addRel(pt[RESPONSE + START], offset, v, 'rp')
-  addRel(pt[RESPONSE + END], offset, v, 'rpe')
-  addRel(pt.domLoading, offset, v, 'dl')
-  addRel(pt.domInteractive, offset, v, 'di')
-  addRel(pt[DOM_CONTENT_LOAD_EVENT + START], offset, v, 'ds')
-  addRel(pt[DOM_CONTENT_LOAD_EVENT + END], offset, v, 'de')
-  addRel(pt.domComplete, offset, v, 'dc')
-  addRel(pt[LOAD_EVENT + START], offset, v, 'l')
-  addRel(pt[LOAD_EVENT + END], offset, v, 'le')
+  handleValue(0, v, 'n')
+  handleValue(pt[UNLOAD_EVENT + START], v, 'u')
+  handleValue(pt[REDIRECT + START], v, 'r')
+  handleValue(pt[UNLOAD_EVENT + END], v, 'ue')
+  handleValue(pt[REDIRECT + END], v, 're')
+  handleValue(pt['fetch' + START], v, 'f')
+  handleValue(pt[DOMAIN_LOOKUP + START], v, 'dn')
+  handleValue(pt[DOMAIN_LOOKUP + END], v, 'dne')
+  handleValue(pt['c' + ONNECT + START], v, 'c')
+  handleValue(pt['secureC' + ONNECT + 'ion' + START], v, 's')
+  handleValue(pt['c' + ONNECT + END], v, 'ce')
+  handleValue(pt[REQUEST + START], v, 'rq')
+  handleValue(pt[RESPONSE + START], v, 'rp')
+  handleValue(pt[RESPONSE + END], v, 'rpe')
+  handleValue(pt.domLoading, v, 'dl')
+  handleValue(pt.domInteractive, v, 'di')
+  handleValue(pt[DOM_CONTENT_LOAD_EVENT + START], v, 'ds')
+  handleValue(pt[DOM_CONTENT_LOAD_EVENT + END], v, 'de')
+  handleValue(pt.domComplete, v, 'dc')
+  handleValue(pt[LOAD_EVENT + START], v, 'l')
+  handleValue(pt[LOAD_EVENT + END], v, 'le')
   return v
 }
 
 // Add Performance Navigation values to the given object
 export function addPN (pn, v) {
-  addRel(pn.type, 0, v, 'ty')
-  addRel(pn.redirectCount, 0, v, 'rc')
+  handleValue(pn.type, v, 'ty')
+  handleValue(pn.redirectCount, v, 'rc')
   return v
 }
 
-export function addRel (value, offset, obj, prop) {
-  var relativeValue
-  if (typeof (value) === 'number' && (value > 0)) {
-    relativeValue = Math.max(Math.round(value - offset), 0)
-    obj[prop] = relativeValue
+function handleValue (value, obj, prop) {
+  if (typeof (value) === 'number' && (value >= 0)) {
+    obj[prop] = Math.round(value)
   }
-  navTimingValues.push(relativeValue)
+  navTimingValues.push(value)
```

Comment:
@patrickhousley When you say "state" of `navTimingValues` do you mean adding (e.g.) `expect(navTimingValues).toEqual(...)` checks similar to the output checks, or something more sophisticated like `navTimingValues` length changing predictably across multiple calls to `addPT` and `addPN`? Looks like those methods are only called once when sending the RUM call.

Reply:
I am specifically talking about unit tests that ensure that when one of the timing methods is called and timings are processed, that the module state is also updated. When, where, and how that exported variable is used is outside the scope of a unit test.

Reply:
Okay, I've suggested tests for that. The only way `navTimingValues` is expected to be useful (as far as I can tell) is to provide a slice of the PT values (not PN), and this usage assumes `addPT` is called first so that the PT values fall in the correct portion of the array (`1, 21`). So technically we wouldn't need to test if `addPN` updates the array, just `addPT`—and maybe `addPN` should not in fact update the array as it does. But I've suggested both tests.